### ISSUE TITLE
quarto gfm + jupyter breaks up a long title

### ISSUE THREAD

**Issue by eyayaw** [quarto-dev/quarto-cli#9579]
### Bug description

Quarto seems to break a long title. I tried putting the title in the YAML and including it as an h1 header. 

### Steps to reproduce

````yaml
---
format: gfm
jupyter: python3
---

# Testing the gradient predictions of the monocentric city model in Addis Ababa

```{python}
print('Hello World')
```
````

### Expected behavior

````md
# Testing the gradient predictions of the monocentric city model in Addis Ababa


``` python
print('Hello World')
```

    Hello World
````

### Actual behavior

````md
# Testing the gradient predictions of the monocentric city model in
Addis Ababa


``` python
print('Hello World')
```

    Hello World


````

### Your environment

- IDE: RStudio 2023.09.0+463 (2023.09.0+463)
- macOS: Sonoma 14.4.1

### Quarto check output

```bash
> quarto check
Quarto 1.4.554
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.554
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/eyayaw/Library/TinyTeX/bin/universal-darwin
      Version: 2024

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.3
      Path: /opt/homebrew/opt/python@3.12/bin/python3.12
      Jupyter: 5.7.2
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.4.0
      Path: /opt/homebrew/Cellar/r/4.4.0/lib/R
      LibPaths:
        - /Users/eyayaw/Library/R/arm64/4.4/library
        - /opt/homebrew/Cellar/r/4.4.0/lib/R/library
      knitr: 1.46
      rmarkdown: 2.26

[✓] Checking Knitr engine render......OK

```

**Comment by cscheid** [comment]
I can't reproduce this on `main`:

````
$ quarto render 9579.qmd
...
$ cat 9579.md


# Testing the gradient predictions of the monocentric city model in Addis Ababa

``` python
print('Hello World')
```

    Hello World
```

**Comment by cscheid** [comment]
I can reproduce this in 1.3 in 1.4, so the fix is to use 1.5 prereleases.
